src/types/mutex.rs

```diff
@@ -0,0 +1,86 @@
+use std::ops::Deref;
+
+/// Wrapper for [`PyMutex`](https://docs.python.org/3/c-api/init.html#c.PyMutex) exposing an RAII interface.
+#[derive(Debug)]
```

Comment:
I think a derived `Debug` implementation here is unsound as it could allow reading while another thread is writing. We should check what the debug implementation for `std::sync::Mutex` prints.

Reply:
Std's Mutex' Debug impl uses `try_lock`, which PyMutex doesn't have. We can have a Debug impl, it just can't print any of its contents.